hack/ci-deploy-release.sh

```diff
@@ -53,6 +53,15 @@ for d in cert-manager{,-cainjector,-webhook}; do
 done
 wait-for-object-creation cert-manager secret/cert-manager-webhook-ca
 
+CRYPTO_KEY_BUFFER_SIZE_MIN=6
+export CRYPTO_KEY_BUFFER_SIZE_MIN
+CRYPTO_KEY_BUFFER_SIZE_MAX=10
+export CRYPTO_KEY_BUFFER_SIZE_MAX
+if [[ -n "${SO_E2E_PARALLELISM:-}" ]]; then
+  CRYPTO_KEY_BUFFER_SIZE_MIN=$(( "${CRYPTO_KEY_BUFFER_SIZE_MIN}" * "${SO_E2E_PARALLELISM}" ))
+  CRYPTO_KEY_BUFFER_SIZE_MAX=$(( "${CRYPTO_KEY_BUFFER_SIZE_MAX}" * "${SO_E2E_PARALLELISM}" ))
+fi
```

Comment:
Starting from an assumption that not overwriting the arguments is desirable for maintainability:

I suggest that we keep the inputs unchanged and instead create dependent variables which are the result of the multiplication by `$SO_E2E_PARALLELISM` having 1 as the default.

WDYT?

Reply:
On a second thought, `SO_E2E_PARALLELISM` in ci-deploy.sh/ci-deploy-release.sh breaks the separation of concerns inbetween these files altogether. Let me rethink this.

Reply:
Also a '0' is a valid argument to `SO_E2E_PARALLELISM`, so the logic that I picked up from #2275 seems broken.

Reply:
@mflendrich I added the defaults and moved things a bit to account for separation of concerns, ptal